GooglePlayInstant/Tests/Editor/QuickDeploy/PlayInstantQuickDeployHttpRequestHelperTest.cs

```diff
@@ -0,0 +1,295 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using GooglePlayInstant.Editor;
+using NUnit.Framework;
+
+namespace GooglePlayInstant.Tests.Editor.QuickDeploy
+{
+    /// <summary>
+    /// Tests for Http Request Helper methods.
+    /// </summary>
+    [TestFixture]
+    public class QuickDeployHttpRequestHelperTest
+    {
+        /*
+         * Testing strategy:
+         *     - Use a local server to inspect contents of the requests sent.
```

Comment:
It feels heavyweight to use a real server to test this code. Using Thread.Sleep() in a test is also a bit of a red flag for me unless there is really no other option.

Right now though I don't see too many good options with the WWW class, which is what we're using for backwards compatibility.

I'm curious to hear what the other reviewers think.

Reply:
Did some research, and for now I think a local server is the more practical and perhaps most feasible option that I found for what we're trying to do because:
#### Can't inspect the parameters that are passed when instantiating the www object. 
After a WWW object is instantiated by passing in the Url, data and headers, Unity does not provide methods or properties that can enable the caller to read the data that has just been passed in
#### Can't mock  WWW/WWform methods involved with sending requests.
As mentioned in the first point, we don't know which methods Unity uses to send the actual HTTP request(we can't even know when it's sent) so it's not possible to mock those methods in order to intercept the data before it is sent to a remote server.
#### Not practical for the code-only plugin to use most mocking frameworks
Since this is not a a feature that is not built in into Unity, one will have to manually move the Mock dll or any dll files for the mocking frameworks being used into our project, which is not in our favor since we want our plugin to be code-only and not have to depend on external things that are not easy to maintain.